how2meet/ui/pages/events.py

```diff
@@ -0,0 +1,139 @@
+from datetime import datetime
+import uuid
+from collections import namedtuple
+
+from nicegui import ui, APIRouter
+import httpx
+
+from ..components.frames import frame
+
+router = APIRouter(prefix="/events",
+                   tags=["events"])
+
+@router.page("/")
+async def events(user: str = None):
+    """List page for all events"""
+    with frame("Events"):
+        # Get the list of events from the API
+        # NOTE: This is a blocking call, so we use an async client
+        async with httpx.AsyncClient() as client:
+            events = await client.get(f"http://localhost:8000/api/events/", timeout=10)
+        try:
+            events = events.json()
+        except:
+            events = []
+
+        async def delete_event(event_id: str):
+            async with httpx.AsyncClient() as client:
+                response = await client.delete(f"http://localhost:8000/api/events/{event_id}", timeout=10)
+                response.raise_for_status()
+            ui.open("/events")
+
+        # Display the list of events
+        for event in events:
```

Comment:
Not blocking, but from UX perspective I was inclined to try and click on the card itself rather than the small buttons. I think nearly all NiceGUI components support an `.on()` method that can do this

Reply:
Yes, this is just a GUI for testing though. This page probably won't even exist in anything like this form in the final app.